tools/ddsrouter_tool/test/compose/validate_subscriber.py

```diff
@@ -0,0 +1,209 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+import re
+import signal
+import subprocess
+
+DESCRIPTION = """Script to validate subscribers output"""
+USAGE = ('python3 validate_subscriber.py -e <path/to/application/executable>'
+         ' [-d]')
+
+PIPE = subprocess.PIPE
+STDOUT = subprocess.STDOUT
+DEVNULL = subprocess.DEVNULL
+
+
+def parse_options():
+    """
+    Parse arguments.
+
+    :return: The arguments parsed.
+    """
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        add_help=True,
+        description=(DESCRIPTION),
+        usage=(USAGE)
+    )
+    required_args = parser.add_argument_group('required arguments')
+    required_args.add_argument(
+        '-e',
+        '--exe',
+        type=str,
+        required=True,
+        help='Path to discovery-server executable.'
+    )
+    parser.add_argument(
+        '-s',
+        '--samples',
+        type=int,
+        default=5,
+        help='Samples to receive.'
+    )
+    parser.add_argument(
+        '-t',
+        '--timeout',
+        type=int,
+        default=5,
+        help='Timeout for the subscriber application.'
+    )
+    parser.add_argument(
+        '--allow-duplicates',
+        action='store_true',
+        help='Allow receive duplicated data.'
+    )
+    parser.add_argument(
+        '-d',
+        '--debug',
+        action='store_true',
+        help='Print test debugging info.'
+    )
+
+    return parser.parse_args()
+
+
+def run(command, timeout):
+    """
+    Run command with timeout.
+
+    :param command: Command to run in list format
+    :param timeout: Timeout for the process
+    :return:
+        - ret_code - The process exit code
+        - stdout - Output of the process
+        - stderr - Error output of the process
+    """
+    ret_code = 0
+
+    proc = subprocess.Popen(command,
+                            stdout=subprocess.PIPE,
+                            universal_newlines=True)
+    try:
+        proc.wait(timeout=timeout)
+    except subprocess.TimeoutExpired:
+        logger.info('Timeout expired. '
+                    'Killing subscriber before receiving all samples...')
+        ret_code = 1
+        proc.send_signal(signal.SIGINT)
+
+    stdout, stderr = proc.communicate()
+
+    return (ret_code, stdout, stderr)
+
+
+def parse_output(data):
+    """
+    Transform the output of the program in a list of received messages.
```

Comment:
```suggestion
    Transform the output of the program in a list of received numbers.
```

Reply:
I think `messages` is more appropriate here as this function list the actual messages.